src/server.c

```diff
@@ -3952,6 +3966,22 @@ int processCommand(client *c) {
     return C_OK;
 }
 
+/* ====================== Error lookup and execution ===================== */
+
+void incrementError(const char* fullerr, size_t namelen) {
+    struct redisError *error = raxFind(server.errors,(unsigned char*)fullerr,namelen);
+    if (error == raxNotFound) {
+        error = zmalloc(sizeof(*error));
+        error->name = zmalloc(namelen+1);
+        memcpy(error->name,fullerr,namelen);
+        error->name[namelen] = '\0';
```

Comment:
you don't need to store the name inside the rax "value".
when iterating on the rax you get the key from the iterator.
considering my previous comments, the error code will always be uppercase. no need to mess with the character casing.

Reply:
fixed in the last commit and squash